spec/PhpSpec/Runner/Maintainer/ErrorMaintainerSpec.php

```diff
@@ -30,10 +30,4 @@ function it_return_true_when_recoverable_level_and_message_match()
         $msg = 'Argument 1 passed to ' . self::class . '::test() must be an instance of string, string given';
         $this->errorHandler(E_RECOVERABLE_ERROR, $msg, 'file', 1)->shouldBe(true);
     }
-
-    function it_throws_error_exception_when_message_not_match()
```

Comment:
Why is this deleted?

Reply:
I think this is not needed anymore in PHP 8.

Could be nice to have confirmation from @alexandrugg or @kelunik ?

Reply:
are you saying it's not possible to test the exception path anymore?

Reply:
I think we need a couple of extra tests to show the behaviour we want from this actually

Reply:
At least in PHP 8, yes.

Reply:
if I test that with `32767 E_ALL` it does throw the exception in both php 7 and 8:
```PHP
function it_throws_error_exception_when_message_not_match()
    {
        $this->shouldThrow(ExampleException\ErrorException::class)
             ->during('errorHandler', [E_ALL, 'error message', 'file', 1]);
    }
```


this being said I can't claim to fully understand this code change 😅 .

Would it be useful to have a test where the `@` [operator](https://www.php.net/manual/en/language.operators.errorcontrol.php) is used to suppress something that would've caused the exception to be thrown in PHP 8, and now with this change it's no longer thrown? Because that's kind of what we're looking to accomplish with this

Reply:
I don't see a reason to delete any test.

Reply:
I deleted the test, maybe I was wrong.

Anyway, despite that, I don't understand the meaning of this test.

If anyone could help, I would be very grateful :)

Reply:
I'll try some stuff so I can better wrap my head around this

Reply:
I opened a separate PR here cause I didn't think I could push to yours: https://github.com/phpspec/phpspec/pull/1378